js/widgets/listview.js

```diff
@@ -211,24 +212,57 @@ return $.widget( "mobile.listview", $.extend( {
 		// by calling addClass() and children() once or twice afterwards. This
 		// can give us a significant boost on platforms like WP7.5.
 
-		for ( itemClass in itemClassDict ) {
-			$( itemClassDict[ itemClass ] ).addClass( itemClass );
-		}
+		for ( dictionaryKey in itemClassDict ) {
 
-		countBubbles.each( function() {
-			$( this ).closest( "li" ).addClass( "ui-li-has-count" );
-		} );
-		if ( countThemeClass ) {
-			countBubbles.not( "[class*='ui-body-']" ).addClass( countThemeClass );
+			// Split the dictionary key back into key classes and extra classes and construct the
+			// _addClass() parameter list
+			this._addClass.apply( this,
+				[ $( itemClassDict[ dictionaryKey ] ) ]
+					.concat( dictionaryKey.split( "|" ) ) );
 		}
 
-		// Deprecated in 1.4. From 1.5 you have to add class ui-li-has-thumb or ui-li-has-icon to the LI.
-		this._addThumbClasses( li );
-		this._addThumbClasses( li.find( ".ui-button" ) );
+		this._addClass(
+			li.filter( function trueIfContextHasBubbleOrHasAnchorThatHasBubble() {
```

Comment:
Would it make sense to extract this out into its own method for code clarity here?